FrontEnd/src/components/SearchListComponent.js

```diff
@@ -229,24 +243,29 @@ function SearchListComponent(props) {
                 /*Der Nutzer wird nur an die TableRow gegeben, wenn entweder nicht gefiltert wird (Für die erste Liste)
                 ODER wenn die Eingabe im Suchfeld dem Parameter des Nutzers entspricht, der im Radiocontrol eingestellt ist
                 (z.B. username des Nutzers entspricht dem im Suchfeld) und zusätzlich die Rolle der im Rollenfilter eingestellten Rolle entspricht */
-                   if(!applyFilter||((dataRecord[selectedSearchOption].toLowerCase().includes(searchFieldValue.toLowerCase())) && (dataRecord[props.filterParameter]===selectedFilterOption||selectedFilterOption==="all"))) {
-                        return (
-                            <TableRow className={dataRecord.rolle==="teacher"?classesCustom.dataRowTeacher:classesCustom.dataRow}>
-                                {/*Es wird hier über alle Einträge des dataRecord iteriert, uns aus jedem Eintrag wird eine Tabellenzelle gebastelt*/}
-                                {Object.values(dataRecord).map((dataRecordField) => {
-                                    return (
-                                        <TableCell align={"center"}>
-                                            <Typography >
-                                                {dataRecordField}
-                                            </Typography>
-                                        </TableCell>
-                                    )
-                                })}
-                                <TableCell>
-                                    {renderButtonsForList(isFirstList, dataRecord)}
-                                </TableCell>
-                            </TableRow>
-                        )
+                   if(!applyFilter||
+                       ((dataRecord[selectedSearchOption].toLowerCase().includes(searchFieldValue.toLowerCase())) &&
```

Comment:
nicht lesbar. Bitte in funktion mit erklärendem Namen auslagern. checkXY() oder so